features/google/src/context/google-context.jsx

```diff
@@ -0,0 +1,188 @@
+import { FeatureStorage } from "@polypoly-eu/feature-storage";
+import { useHistory, useLocation } from "react-router-dom";
+import { importData } from "../model/importer";
+
+export const GoogleContext = React.createContext();
+
+//used until real storage is loaded
+const fakeStorage = {
+    files: null,
+    refreshFiles: async () => null,
+    readFile: async () => null,
+    removeFile: async () => {},
+};
+
+class RefreshFilesError extends Error {
+    constructor(cause) {
+        super("Failed to refresh files");
+        this.name = "RefreshFilesError";
+        this.cause = cause;
+    }
+}
+
+class FileImportError extends Error {
+    constructor(cause) {
+        super("Failed to import file");
+        this.name = "FileImportError";
+        this.cause = cause;
+    }
+}
+
+class FileSelectionError extends Error {
+    constructor(cause) {
+        super("Failed to select file");
+        this.name = "FileSelectionError";
+        this.cause = cause;
+    }
+}
+
+function updatePodNavigation(pod, history, handleBack, location) {
+    pod.polyNav.actions = {
+        back: () => handleBack(),
+    };
+    history.length > 1 &&
+    location.pathname !== "/overview" &&
+    location.pathname !== "/import"
+        ? pod.polyNav.setActiveActions(["back"])
+        : pod.polyNav.setActiveActions([]);
+}
+
+export const GoogleContextProvider = ({ children }) => {
+    const [pod, setPod] = useState(null);
+    const [files, setFiles] = useState(null);
+    const [storage, setStorage] = useState(fakeStorage);
+    const [globalError, setGlobalError] = useState(null);
+    const [selectedFile, setSelectedFile] = useState(null);
+    const [googleAccount, setGoogleAccount] = useState(null);
+
+    const location = useLocation();
+    const initPod = async () => await window.pod;
+    const history = useHistory();
+
+    function handleBack() {
+        if (history.length > 1) {
+            history.goBack();
+        }
+    }
+
+    async function runWithLoadingScreen(task) {
+        setFiles(null);
+        await task();
+        refreshFiles();
+    }
+
+    const handleSelectFile = async () => {
+        const { polyNav } = pod;
+        runWithLoadingScreen(async function () {
+            try {
+                setSelectedFile(await polyNav.pickFile("application/zip"));
+            } catch (error) {
+                setGlobalError(new FileSelectionError(error));
+            }
+        });
+    };
+
+    const handleRemoveFile = (fileID) => {
+        return storage.removeFile(fileID);
+    };
+
+    const handleImportFile = async () => {
+        if (!selectedFile) return;
+        const { polyOut } = pod;
+        console.log("handleImportFile: ", selectedFile.url);
+        runWithLoadingScreen(async function () {
+            try {
+                await polyOut.importArchive(selectedFile.url);
+                setSelectedFile(null);
+            } catch (error) {
+                setGlobalError(new FileImportError(error));
```

Comment:
Or make it clear how we want to do this.
We could do: 

1. Identify what you want to make common in your Code
2. Make PR's for these things
3. Make tests
4. Get feedback
5. Merge
6. Code the thing you initially wanted to code with the now in poly-look or where ever shared building blocks

I just think the productivity would suffer under this overhead to be completely honest

Reply:
Think about the time you will save refactoring later, or after the reviews.
I'm not so adamant about error handling, either, because as I say it's so intricately tied to the code that it would be difficult to factor out. The issue here is different: we're opting for using a global state to store the error, and I'm not entirely clear where that is actually handled. If it ends up in the view, that's an option, and it's valid as first approximation, but we need to have a very clear understanding of error handling and the corresponding data flows.